core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -331,7 +331,7 @@ public HttpResponse doConfigSubmit(StaplerRequest req, StaplerResponse rsp) thro
         j.clouds.replace(this, result);
         j.save();
         // take the user back to the cloud top page.
-        return FormApply.success(".");
+        return FormApply.success("..");
```

Comment:
Apply generally will not be change the page, I would guess the save 404 is the actual save failing after the rename as the endpoint has gone.

Reply:
> I would guess the save 404 is the actual save failing after the rename as the endpoint has gone

Yes, that is exactly what's happening. If that's the case, then I'm going to mark this PR ready to review.

Reply:
Like for projects the rename should be switched to a dedicated page, accessible via the sidepanel instead of allowing to change the name as part of the configure page. That would solve the issue with the 404 after an apply with a rename

Reply:
@mawinter69 Could that be considered as a follow up ticket for an enhancement while this ticket could be considered fixing the bug?

Reply:
https://github.com/jenkinsci/jenkins/pull/8310/files#r1278343520

This PR doesn't work at all for me.

Tested with Azure VM Agents.

Rename cloud, click save,
page I was on before save:
http://localhost:8080/jenkins/cloud/cloud-5/configure
after save:
http://localhost:8080/jenkins/cloud/cloud-5/

end up on 404
<img width="744" alt="image" src="https://github.com/jenkinsci/jenkins/assets/21194782/0a200336-1dde-49f4-8619-2500fbc7356a">

----

Click Apply after renaming the cloud.
Apply works fine

Click apply again

404 popup:

<img width="1123" alt="image" src="https://github.com/jenkinsci/jenkins/assets/21194782/dea8811b-3dac-4079-8abd-f2cfa465cd61">

----

so no I don't think a follow up ticket is sufficient

Reply:
One problem with a dedicated rename page is that most clouds have the name as part of their config page (Mesos plugin doesn't have it and uses a fixed name, which results in a cloudByIndex url when there is more than one). So all cloud provider plugins would need to be adjusted as well to ensure the problem can't appear again.

Reply:
A rename page could redirect to the new name though if it can tell that the cloud name is used in the url

Reply:
Using dedicated rename page